src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
      *
-     * @for LegModel
-     * @method getTopAltitude
-     * @return {number}
+     * @for Fms
+     * @method _initializeArrivalAirport
+     * @param destinationIcao {string} ICAO identifier specified by spawn pattern
+     * @private
      */
-    getTopAltitude() {
-        const maxAltitudeFromLegs = _map(this.legCollection, (leg) => leg.getProcedureTopAltitude());
+    _initializeArrivalAirport(destinationIcao) {
+        if (destinationIcao === '') {
+            return;
+        }
 
-        return Math.max(...maxAltitudeFromLegs);
+        this.arrivalAirportModel = AirportController.airport_get(destinationIcao);
     }
 
     /**
-     * Collects the `.getProcedureBottomAltitude()` value from each `LegModel`
-     * in the `#legCollection`, then finds and returns the lowest value
+     * Initialize #arrivalRunwayModel
      *
-     * @for LegModel
-     * @method getBottomAltitude
-     * @return {number}
+     * @for Fms
+     * @method _initializeArrivalRunway
+     * @private
      */
-    getBottomAltitude() {
-        const valueToExclude = INVALID_NUMBER;
-        const minAltitudeFromLegs = _without(
-            _map(this.legCollection, (leg) => leg.getProcedureBottomAltitude()),
-            valueToExclude
-        );
+    _initializeArrivalRunway() {
+        if (this.arrivalAirportModel === null) {
+            return;
+        }
 
-        return Math.min(...minAltitudeFromLegs);
-    }
+        const arrivalRunwayName = this._routeModel.getArrivalRunwayName();
 
-    /**
-     * Get the flight plan route string in dot notation
-     *
-     * @for Fms
-     * @method getFlightPlanRouteStringWithDots
-     * @return {string}
-     */
-    getFlightPlanRouteStringWithDots() {
-        const currentRouteSegments = _map(this.legCollection, (legModel) => legModel.routeString);
-        const previousAndCurrentRouteSegments = this._previousRouteSegments.concat(currentRouteSegments);
+        if (arrivalRunwayName === null) {
+            return this.setArrivalRunway(this.arrivalAirportModel.arrivalRunwayModel);
+        }
+
+        const arrivalRunwayModel = this.arrivalAirportModel.getRunway(arrivalRunwayName);
 
-        return previousAndCurrentRouteSegments.join('..').toUpperCase();
+        this.setArrivalRunway(arrivalRunwayModel);
     }
 
     /**
-     * Get the flight plan route string with legs separated by spaces
-     *
-     * This is primarily meant for use in the `StripViewModel`.
+     * Initialize #departureAirportModel
      *
      * @for Fms
-     * @method getFlightPlanRouteStringWithSpaces
-     * @return {string}
+     * @method _initializeDepartureAirport
+     * @param originIcao {string} ICAO identifier specified by spawn pattern
+     * @private
      */
-    getFlightPlanRouteStringWithSpaces() {
-        return this.getFlightPlanRouteStringWithDots().replace(REGEX.DOUBLE_OR_SINGLE_DOT, ' ');
+    _initializeDepartureAirport(originIcao) {
+        if (originIcao === '') {
```

Comment:
would `airportIcao` ever not be empty and invalid? would it ever not be a string?